libpod/container_internal.go

```diff
@@ -1708,6 +1708,14 @@ func (c *Container) mountStorage() (_ string, deferredErr error) {
 	tz := c.Timezone()
 	if tz != "" {
 		timezonePath := filepath.Join("/usr/share/zoneinfo", tz)
+
+		// Allow using TZDIR per:
+		// https://sourceware.org/git/?p=glibc.git;a=blob;f=time/tzfile.c;h=8a923d0cccc927a106dc3e3c641be310893bab4e;hb=HEAD#l149
+		tzdir := os.Getenv("TZDIR")
+		if tzdir != "" {
+			timezonePath = filepath.Join(tzdir, tz)
```

Comment:
Wait.... this seems wrong to me, it's completely different from the append-path approach in your c-c PR. And I don't see any other basic utils if TZDIR is missing. I'm still investigating, but for now, let's block this

/hold

(Oh, also, what Dan said. We don't want two inconsistent implementations. But that might be a bigger problem)

Reply:
Yeah so the question to me would be: Do we want to mimic the glibc behavior and use `$TZDIR` as override or do we want to place it as a _higher priority path_? I'm leaning towards the override.

Reply:
Yeah so the question to me would be: Do we want to mimic the glibc behavior and use `$TZDIR` as override or do we want to place it as a _higher priority path_?

Reply:
The [glibc code you linked to](https://sourceware.org/git/?p=glibc.git;a=blob;f=time/tzfile.c;h=8a923d0cccc927a106dc3e3c641be310893bab4e;hb=HEAD#l149) implements it as an override; the wording in `tzset(3)` seems to confirm:
>TZDIR  If this variable is set its value takes precedence over the system configured timezone database directory path.

Okay, so it's c-c that is wrong. It squicks me out to have podman actually error out, instead of just ignoring the error, but there's precedent (`podman run --tz=Foo/Bar` also fails), so okay.

Please fix it so it actually works though (my `Abc/Def` test above).

Thanks for your PR and for providing good reference sources.

Reply:
Alright, following-up in c/common: https://github.com/containers/common/pull/1774